trainer/lib/trainer/test_parser.rb

```diff
@@ -202,7 +202,11 @@ def parse_xcresult(path, output_remove_retry_attempts: false)
       path = Shellwords.escape(path)
 
       # Executes xcresulttool to get JSON format of the result bundle object
-      result_bundle_object_raw = execute_cmd("xcrun xcresulttool get --format json --path #{path}")
+      # Hotfix: From Xcode 16 beta 3 'xcresulttool get --format json' has been deprecated; '--legacy' flag required to keep on using the command
+      xcresulttool_cmd = "xcrun xcresulttool get --format json --path #{path}"
+      xcresulttool_cmd << ' --legacy' if `xcrun --version`.gsub('xcrun version ', '').to_f >= 70.0 # Version of xcrun bundled in Xcode 16 beta 3
```

Comment:
Could you use `xcrun xcresulttool version` instead to check backward compatibility?
This behavior depends on `xcresulttool` version, not `xcrun` version.

`xcresulttool version 23021, format version 3.53 (current)`

`format version` can be used.

It's better to extract the method to parse `xcresulttool` version

```ruby
private
def xcresulttool_version
  result.match(/format version (?<format_version>\d+\.\d+)/)
end
```

Reply:
```
$ DEVELOPER_DIR=/Applications/Xcode_15.4.app xcrun xcresulttool version
xcresulttool version 22608, format version 3.49 (current)

$ DEVELOPER_DIR=/Applications/Xcode_16_beta.app xcrun xcresulttool version
xcresulttool version 23019.2, format version 3.53 (current)

$ DEVELOPER_DIR=/Applications/Xcode_16_beta_2.app xcrun xcresulttool version
xcresulttool version 23019.2, format version 3.53 (current)

$ DEVELOPER_DIR=/Applications/Xcode_16_beta_3.app xcrun xcresulttool version
xcresulttool version 23021, format version 3.53 (current)
```

Thanks for the information. I have never known there is `xcresulttool version` command and just accepted suggestion from other. 
Pinpointing `xcresulttool` version would be the best answer. Thanks again!